segmentation/utils/synthetic_spatial_datagen.py

```diff
@@ -0,0 +1,167 @@
+import numpy as np
+import xarray as xr
+import pandas as pd
+import skimage.measure
+import scipy
+import os
+import xarray as xr
+import copy
+
+from segmentation.utils import spatial_analysis_utils as sau
+from scipy.spatial.distance import cdist
+from skimage.measure import label
+
+
+def direct_init_dist_matrix(num_A=100, num_B=100, num_C=100,
+                            ab_dist_mean=100, ab_dist_var=1, ac_dist_mean=20, ac_dist_var=1,
+                            distr_AB={'mean': 100, 'var': 1}, distr_AC={'mean': 20, 'var': 1},
+                            seed=None):
+    """
+    This function will return a random dist matrix such that the distance between cells
+    of types A and B are overall larger than the distance between cells of types A and C
+
+    Each row and column representing a cell.
+    We generate the points using Gaussian distributions
+    Ideally, the parameters for A to B distances will be set such that they produce a lower range of values
+    than A to C distances.
+
+    Will return a random distance matrix as well as the dictionary of associated cell: label IDs
+    The above is generated from the generate_labels function
+
+    Args:
+        num_A: the number of A cells we wish to generate. Default 100
+        num_B: the number of B cells we wish to generate. Default 100
+        num_C: the number of C cells we wish to generate. Default 100
+        distr_AB: if specified, will be a dict listing the mean and variance of the Gaussian distribution
+            we wish to generate numbers from. Default mean=100 and var=1
+        distr_AC: similar to dist_AB. Default mean=20 and var=1
+        seed: whether to fix the random seed or not. Useful for testing.
+            Should be a specified integer value. Default 42.
+    """
+
+    # set the mean and variance of the Gaussian distributions of both AB and AC distances
+
+    mean_ab = distr_AB['mean']
+    var_ab = distr_AB['var']
+
+    mean_ac = ac_dist_mean
+    var_ac = ac_dist_var
+
+    # set random seed if set
+    if seed:
+        np.random.seed(seed)
+
+    # we initialize the random distances across different types of points
+    # note that we don't really care about aa, bb, bc, or cc, so we
+    # initialize those to garbage. We do need them for a proper
+    # distance matrix format, however.
+    random_aa = np.random.normal(0, 1, (num_A, num_A))
+    random_ab = np.random.normal(mean_ab, var_ab, (num_A, num_B))
+    random_ac = np.random.normal(mean_ac, var_ac, (num_A, num_C))
+    random_bb = np.random.normal(0, 1, (num_B, num_B))
+    random_bc = np.random.normal(0, 1, (num_B, num_C))
+    random_cc = np.random.normal(0, 1, (num_C, num_C))
+
+    # create each row one-by-one first
+    # we need to correct each aa, bb, and cc matrix to ensure symmetry
+    first_row = np.concatenate(((random_aa + random_aa.T) / 2, random_ab, random_ac), axis=1)
```

Comment:
Perhaps we can find a different name besides "row", since it's not a single row? This was confusing me when I tried to figure out what's going on here.